### PR TITLE
Switch to Noble for Rolling.

### PR DESCRIPTION
**Pull request by clalancette**

### CODE REVIEW

**Comment by christophebedard** [comment]
```
  /usr/bin/bash -c source /opt/ros/rolling/setup.sh && colcon coveragepy-result --packages-select libstatistics_collector --verbose --coverage-report-args -m
  /usr/local/lib/python3.12/dist-packages/colcon_core/argument_parser/type_collector.py:9: UserWarning: 'colcon_core.argument_parser.type_collector.TypeCollectorDecorator' has been deprecated
    class TypeCollectorDecorator(ArgumentParserDecorator):
  /usr/local/lib/python3.12/dist-packages/colcon_core/argument_parser/type_collector.py:62: UserWarning: 'colcon_core.argument_parser.type_collector.SuppressTypeConversions' has been deprecated, use 'colcon_core.argument_parser.action_collector.SuppressTypeConversions' instead
    class SuppressTypeConversions:
  usage: colcon [-h] [--log-base LOG_BASE] [--log-level LOG_LEVEL]
                {build,coveragepy-result,extension-points,extensions,graph,info,lcov-result,list,metadata,mixin,test,test-result,version-check}
                ...
  colcon: error: unrecognized arguments: -m
```

This was just fixed and I just created a new release, so I'll try to re-trigger the job.

---

> Warning: Unexpected input(s) 'colcon-mixin-name'

This needs to be replaced by `colcon-defaults` (https://github.com/ros-tooling/action-ros-ci/pull/591), e.g.: https://github.com/ros2/ros2_tracing/blob/c0e59c442e61f6c638f06a48a00ecf10913442fb/.github/workflows/test.yml#L76

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,13 +13,19 @@ jobs:
...
     container:
-      image: rostooling/setup-ros-docker:ubuntu-jammy-latest
+      image: rostooling/setup-ros-docker:ubuntu-noble-latest
     steps:
     - uses: ros-tooling/action-ros-ci@v0.3
       with:
         package-name: libstatistics_collector
-        colcon-mixin-name: coverage-gcc
-        colcon-mixin-repository: https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
+        colcon-defaults: |
```
Comment (christophebedard) [inline .github/workflows/test.yml:21]: I think the text below needs to be indented with 2 spaces: https://github.com/ros-tooling/libstatistics_collector/actions/runs/8882584262.

File: .github/workflows/autoapprove.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
   # Dependabot will automatically merge minor version upgrades
   # (see .dependabot/config.yml for more info).
   auto-approve-dependabot:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (christophebedard) [inline .github/workflows/autoapprove.yml]: The `ubuntu-24.04` runner is not available yet. The solution is to use `ubuntu-latest` and use Docker with `ubuntu:24.04`. See the tips here: https://github.com/ros-tooling/setup-ros/releases/tag/0.7.2

Comment (clalancette) [reply .github/workflows/autoapprove.yml]: I kind of figured that.  It's just that for some reason, GitHub refuses to run *any* of the workflow jobs in this PR.  Sigh.  I'll revert all of that.

Comment (christophebedard) [reply .github/workflows/autoapprove.yml]: I think there's also just a lot of queued jobs that don't get canceled when pushing new commits.

**Comment by clalancette** [comment]
Whooppee!  It works, finally.  @christophebedard I could use review when you have a minute.

**Review by christophebedard — APPROVED** [review]

**Comment by clalancette** [comment]
Thanks, appreciated!  With that, I'm going to merge this in since it only touched the github jobs and not the code at all.

**Pull request merged**